src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+//! This is the config struct used to pass in configurations into the channel. This is kept by the channel manager
+//! channel_limits is only passed in as a reference and only used when accepting or opening a channel
+//! channel options is cloned into channel because those are all specified unique channel settings. It uses channel managers one as a base
+
+/// This is the main user config
+/// This struct should contain all user customizable options as this is passed to the channel to be accessed
+#[derive(Clone, Debug)]
+pub struct UserConfig{
+	/// optional user specified channel limits
+	/// These are only used on startup of channels, and are referenced to a single instance
+	pub channel_limits : ChannelHandshakeLimits,
+	/// channel options are unique to each channel
+	pub channel_options : ChannelConfig,
+}
+
+impl UserConfig {
+	///default constructor, calls ChannelOptions and ChannelLimits constructors
+    pub fn new() -> Self{
+        UserConfig {
+            channel_limits : ChannelHandshakeLimits::new(),
+			channel_options : ChannelConfig::new(),
+        }
+    }
+}
+
+/// This struct contains all the optional bolt 2 channel limits.
+/// If the user wants to check a value, the value needs to be filled in, as by default they are not checked
+#[derive(Copy, Clone, Debug)]
+pub struct ChannelHandshakeLimits{
+	/// minimum allowed funding_satoshis
```

Comment:
Same for all the per-field comments here - can you describe what they actually mean, not just the variable name?